### ISSUE TITLE
Problem with the horizontal scrolling in the datagrid

### ISSUE THREAD

**Issue by alekseyLymarev** [IgniteUI/igniteui-angular#2068]
### Description  
Hello!

When i 'm using datagrid occurres a problem with a horizontal scrolling. Datagrid freezes on scroll to the left or to the right.

Grid contains 28 columns and approximately 120 rows.

I'm using ng-template to render custom contents of cells. When im not using ng-template - there are almost no freeze, while scrolling. But when am using ng-template (even if i just interpolate variable in it, like this {{value}}) component begining to lag hardly, even if there are 20 rows. 

Please tell what could be the problem and how i can fix it?

 * igniteui-angular version: 6.1.0
 * browser: Chrome v62.0.3202.94

**Comment by kdinev** [comment]
@alekseyLymarev Does this happen only with keyboard scroll, or in general: keyboard, mouse, touch?

**Comment by kdinev** [comment]
@MayaKirova Please check if this is related/same as #1923

**Comment by MayaKirova** [comment]
@alekseyLymarev @kdinev 
I have investigate this and it seems to indeed be related to an already existing issue - #1923.
In general the issue seems to be that when templates change for existing embedded view (which we re-use as part of our virtualization implementation) Angular seems to take a very long time to detect changes on them and render the new templates.
We have already started working on possible way to improve the performance in such scenarios.

**Comment by MayaKirova** [comment]
@alekseyLymarev 
By the way, if you’re using the same template for different columns, for instance if you want to wrap/style the content in some way that should be applied to all or for multiple columns, you could re-use the same template instance instead of creating new ones for each. This should help with the performance.
For example, instead of creating new ones for each column:

```
<igx-grid #grid1 [data]="localData" width="960px" height="550px">
    <igx-column *ngFor="let c of columns" [field]="c.field" [header]="c.field" [width]="c.width">
        <ng-template igxCell let-val>
            Templated Value: {{val}}
        </ng-template>
      </igx-column>
</igx-grid>
```
You can define a single one:
```
<igx-grid #grid1 [data]="localData"width="960px" height="550px" (onColumnInit)="initColumns($event)">
    <igx-column *ngFor="let c of columns" [sortable]="true" [filterable]="true" [field]="c.field" [header]="c.field" [width]="c.width">
      </igx-column>
</igx-grid>

<ng-template #template let-val>
    Templated Value: {{val}}
  </ng-template>
```
And set for all columns to use the same instance when the columns are initialized:
```
  @ViewChild('template', { read: TemplateRef })
  protected template: TemplateRef<any>;

  initColumns(col: any) {
    col.bodyTemplate = this.template;
}
```

I’ve attached a small sample with both approaches so that you can compare the performance when the templates are reused:
[igxGrid.zip](https://github.com/IgniteUI/igniteui-angular/files/2236008/igxGrid.zip)

Hope this helps a bit in the meantime.

**Comment by tachojelev** [comment]
The #2252 and #2255 PRs fix this issue and #2251.
However they do **not** fix the other mentioned issue - #1923 , since it was not addressed in the PRs.

**Comment by MayaKirova** [comment]
@tachojelev For #1923 additional optimizations need to be done around the cell components event handling.
